DigitalLearningSolutions.Web/Controllers/TrackingSystem/CourseSetup/ManageCourseController.cs

```diff
@@ -41,5 +42,51 @@ public IActionResult Index(int customisationId)
 
             return View(model);
         }
+
+        [HttpGet]
+        [Route("{customisationId:int}/EditCourseOptions")]
+        public IActionResult EditCourseOptions(int customisationId)
+        {
+            var centreId = User.GetCentreId();
+            var categoryId = User.GetAdminCategoryId()!;
+
+            var courseDetails = courseDataService.GetCourseDetailsForAdminCategoryId(
+                customisationId,
+                centreId,
+                categoryId.Value
+            );
+
+            if (courseDetails == null)
+            {
+                return NotFound();
+            }
+
+            var model = new EditCourseOptionsViewModel(courseDetails);
+            return View(model);
+        }
+
+        [HttpPost]
+        [Route("{customisationId:int}/EditCourseOptions")]
+        public IActionResult EditCourseOptions(EditCourseOptionsViewModel editCourseOptionsViewModel)
+        {
+            var centreId = User.GetCentreId();
+            var courseDetails = new CourseDetails
+            {
+                CustomisationId = editCourseOptionsViewModel.CustomisationId,
+                CentreId = centreId,
+                Active = editCourseOptionsViewModel.Active,
+                SelfRegister = editCourseOptionsViewModel.AllowSelfEnrolment,
+                HideInLearnerPortal = editCourseOptionsViewModel.HideInLearningPortal,
+                DiagObjSelect = editCourseOptionsViewModel.DiagnosticObjectiveSelection
+            };
+
+            var updateSuccessful = courseDataService.UpdateCourseOptions(courseDetails);

```

Comment:
We shouldn't update the course if the user's category id doesn't match and isn't 0.